### ISSUE TITLE
water from 

### ISSUE THREAD

**Issue by orioncohen** [openbabel/openbabel#2431]
- [x] I believe this to be a bug with Open Babel
- [] This is a feature request

## Environment Information
Open Babel version: 3.1.1
Operating system and version: macOS Catalina 10.15.7

## Expected Behavior
Generating H<sub>2</sub>O from SMILE "O" should yield a Moelcule with a single residue.

## Actual Behavior
H<sub>2</sub>O is two residues!

## Steps to Reproduce
```python
from openbabel import pybel

mol = pybel.readstring("smi", "O")
mol.addh()
mol.make3D()

len(mol.atoms ==3)  # True
len(mol.residues) == 2  # True
```
Now `len(mol.residues) == 2`!

Is this the intended behavior? If not, is there a hot fix to yield only one residue? The fact that the mol ends up being two residues is inconvenient for some MD IO I am working on.

**Comment by fredrikw** [comment]
I cannot reproduce this (on OB 3.1.0 on wsl2, should be feature matched to 3.1.1)
```
>>> from openbabel import pybel
>>> mol = pybel.readstring("smi", "O")
>>> mol.addh()
>>> mol.make3D()
>>> len(mol.atoms) == 3  # True
True
>>> len(mol.residues) == 2  # True
False
```

In fact, mol.residues is an empty array at this point for me.

**Comment by orioncohen** [comment]
Hi @fredrikw, thank you for taking a look at this. I misattributed the error in my initial code. The water doesn't actually get two residues until I attempt to write the file to a 'pdb'. 

```python
from openbabel import pybel

mol = pybel.readstring("smi", "O")
mol.addh()
mol.make3D()
mol.write(format="pdb", filename="H2O")  # added this line

len(mol.atoms) == 3  # True
len(mol.residues) == 2  # True
```

Oddly, if I write `mol` to other file formats, like "mol", `mol` only has a single residue.

**Comment by ghutchis** [comment]
Yeah, I'm not sure where the error is, but the PDB file includes this:
```
HETATM    1  O   HOH     1       0.956   0.096  -0.020  1.00  0.00           O  
HETATM    2  H   HOH     0       1.924   0.095   0.026  1.00  0.00           H  
HETATM    3  H   HOH     0       0.677   0.091   0.908  1.00  0.00           H  
```

I suspect the problem is in the chain perception.. which is about the worst code to debug.

**Comment by orioncohen** [comment]
Thanks all. I've solved my problem by using "mol" as an intermediate file format instead of "pdb", so this is no longer an issue for me personally. I'll leave it to the discretion of the devs whether or not this should remain open.
